### PR TITLE
PIM-5788: add details about import / export migrations

### PR DESCRIPTION
**Pull request by nidup**
Complete the upgrade.md doc (more to come)

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       |
| Added Behats                      |
| Changelog updated                 |
| Review and 2 GTM                  |
| Micro Demo to the PO (Story only) |
| Migration script                  |
| Tech Doc                          |

### CODE REVIEW

File: UPGRADE-1.6.md [hunk 1]
```diff
@@ -204,7 +205,7 @@ You can find further details by visiting our technical documentation [here](http
 
 ### Global updates for any projects
 
-#### Update references to Pim\Bundle\CatalogBundle\AttributeType\AbstractAttributeType
+#### Update references to moved `Pim\Bundle\CatalogBundle\AttributeType\AbstractAttributeType` constants
```
Comment (aRn0D) [inline UPGRADE-1.6.md]: `to the moved`? the next chapter `to the removed`

Comment (nidup) [inline UPGRADE-1.6.md]: we moved constant from abtract to interface

File: UPGRADE-1.6.md [hunk 2]
```diff
@@ -579,17 +631,80 @@ For instance, if you have a custom csv export using the same parameters than our
             - { name: akeneo_batch.job.job_parameters.constraint_collection_provider }
 ```
 
-If your job needs extra runtime parameters, you have to declare your own classes.
+Case 2 - You wrote a custom import (or export) which needs default parameters and few extra runtime parameters.
+
+You can take inspiration of the native csv product export which use the simple csv export parameters and add its own parameters.
+
+For instance, for the csv product export DefaultValuesProviderInterface :
```
Comment (aRn0D) [inline UPGRADE-1.6.md]: `DefaultValuesProviderInterface`

File: UPGRADE-1.6.md [hunk 3]
```diff
@@ -579,17 +631,80 @@ For instance, if you have a custom csv export using the same parameters than our
...
+}
+```
+
+The same strategy can be used for `ConstraintCollectionProviderInterface`.
+
+Case 3 - You wrote a custom import (or export) which needs its very own runtime parameters.
+
+You have to declare your own services and classes implementing `DefaultValuesProviderInterface` and `ConstraintCollectionProviderInterface`.
+
+Finally, you can remove useless getter / setter and related parameters from your StepElements.
```
Comment (aRn0D) [inline UPGRADE-1.6.md]: and the method `getConfigurationFields`

File: UPGRADE-1.6.md [hunk 4]
```diff
@@ -601,13 +716,75 @@ For instance, if you have a custom csv export using the same parameters than our
...
+           - { name: pim_enrich.view_element, type: pim_import_export_jobInstance.export.form_tab, position: 100 }
+
+```
+
+#### Update access to the configuration in your custom Reader, Processor, Writer by using the JobParameters
+
+The JobParameters allows to store the runtime parameters.
 
-We've extracted the template concern from JobInterface.
+When running a job, the JobParameters is configured, validated and injected in the JobExecution.
```
Comment (aRn0D) [inline UPGRADE-1.6.md]: missing `...` same at the next line

File: UPGRADE-1.6.md (hunk 2 above)
Comment (nidup) [inline UPGRADE-1.6.md]: fixed

**Pull request merged**